lang/LangPrimSource/PyrArrayPrimitives.cpp

```diff
@@ -2369,12 +2383,142 @@ int prArrayUnlace(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+extern bool compiledOK;
+PyrSymbol* symb_unixCmdAction;
+
+static void string_popen_thread_func(struct sc_process *process)
+{
+	FILE *stream = process->stream;
+	pid_t pid = process->pid;
+	char buf[1024];
+
+	while (process->postOutput) {
+		char *string = fgets(buf, 1024, stream);
+		if (!string) break;
+		postText(string, strlen(string));
+	}
+
+	int res;
+	res = sc_pclose(stream, pid);
+	res = WEXITSTATUS(res);
+
+	if(process->postOutput)
+		postfl("RESULT = %d\n", res);
+
+	free(process);
```

Comment:
btw2: c++11 has `std::unique_ptr`, a movable smart pointer, which avoids the need to do manual memory management (i know this code is old)